common/web/keyboard-processor/src/keyboards/loaders/nodeKeyboardLoader.ts

```diff
@@ -23,15 +23,21 @@ export class NodeKeyboardLoader extends KeyboardLoaderBase {
   }
 
   protected loadKeyboardInternal(uri: string): Promise<Keyboard> {
+    // `fs` does not like 'file:///'; it IS "File System" oriented, after all, and wants a path, not a URI.
+    if(uri.indexOf('file:///') == 0) {
+      uri = uri.substring('file:///'.length);
+    }
+
+    let script;
+    try {
+      script = new vm.Script(fs.readFileSync(uri).toString());
+    } catch (err) {
+      return Promise.reject(new Error(KeyboardLoaderBase.missingErrorMessage(uri)));
+    }
     try {
-      // `fs` does not like 'file:///'; it IS "File System" oriented, after all, and wants a path, not a URI.
-      if(uri.indexOf('file:///') == 0) {
-        uri = uri.substring('file:///'.length);
-      }
-      const script = new vm.Script(fs.readFileSync(uri).toString());
       script.runInContext(this.harness._jsGlobal);
     } catch (err) {
-      return Promise.reject(err);
+      return Promise.reject(new Error(KeyboardLoaderBase.scriptErrorMessage(uri)));
     }
```

Comment:
It'd be good if the actual error message thrown (`err`) was not lost in these two exception handlers.

Reply:
That's very fair.  I was looking for a 'simple' way to do that and came up short in my first pass, but I could always take an extra look.